crates/matrix-sdk/src/widget/matrix.rs

```diff
@@ -54,17 +52,17 @@ impl MatrixDriver {
     }
 
     /// Requests an OpenID token for the current user.
-    pub(crate) async fn get_open_id(&self) -> HttpResult<OpenIdResponse> {
+    pub(crate) async fn get_open_id(&self) -> Result<OpenIdResponse, Error> {
         let user_id = self.room.own_user_id().to_owned();
-        self.room.client.send(OpenIdRequest::new(user_id), None).await
+        self.room.client.send(OpenIdRequest::new(user_id), None).await.map_err(Error::Http)
     }
 
     /// Reads the latest `limit` events of a given `event_type` from the room.
     pub(crate) async fn read_message_like_events(
         &self,
         event_type: MessageLikeEventType,
         limit: u32,
-    ) -> Result<Vec<Raw<AnyTimelineEvent>>> {
+    ) -> Result<Vec<Raw<AnyTimelineEvent>>, Error> {
```

Comment:
nit: can be reverted, because `Result<T>` is a synonym for `Result<T, crate::Error>` if the error type isn't specified.